src/main/java/states/ReadState.java

```diff
@@ -89,7 +96,16 @@ public State execute() {
 						break;
                     }
 
-                } catch (InvalidPacketException e) {
+                } catch(IOException ioE) {
+                	if(ioE.getMessage().contains("Not enough usable space")) {
+        				LOG.logQuiet("Not enough usable disk space");
+                		ErrorMessage msg = new ErrorMessage(ErrorType.DISK_FULL, "Not enough free space on disk");
+        				socket.sendMessage(msg, recvSocketAddress);
```

Comment:
This is the client, are we supposed to be sending packets like this to the server?